### PR TITLE
Limiting python version on both arena setup and CICD tests and package release

### PR DESCRIPTION
**Pull request by alexpalms**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
     long_description = (Path(__file__).parent / "README.md").read_text(),
     long_description_content_type="text/markdown",
     license='Custom',
+    python_requires='>=3.7, <3.11',
```
Comment (discordianfish) [inline setup.py:32]: Should we limit this here? Using python >3.10 should be fine unless you use tensorflow 2.x, right?

**Review by discordianfish — APPROVED** [review]
See comment but LGTM in general. If I miss something and we need to indeed limit the python version in the pip package as well, feel free to merge this.

File: setup.py [hunk 2]
```diff
@@ -43,7 +44,7 @@
             'opencv-contrib-python>=4.4.0.42',
             'opencv-python>=4.4.0.42',
             'grpcio',
-            'diambra-engine',
+            'diambra-engine>2.1.x', # To make sure both protobuf <3.20 (TF 2.x current limitation) and >=3.20 are supported
```
Comment (alexpalms) [inline setup.py:47]: @discordianfish I am facing a problem with CICD here:
1) If I use `'diambra-engine>2.1.x'` CICD for tests works but release fails
2) If I use `'diambra-engine>2.1'` CICD for tests fails but release works

Locally, when I install the packages, both works... so really no idea.

File: setup.py [hunk 3]
```diff
@@ -43,7 +44,7 @@
             'opencv-contrib-python>=4.4.0.42',
             'opencv-python>=4.4.0.42',
             'grpcio',
-            'diambra-engine',
+            'diambra-engine>2.1.0rc5', # To make sure both protobuf <3.20 (TF 2.x current limitation) and >=3.20 are supported
```
Comment (alexpalms) [inline setup.py:47]: @discordianfish here I am encountering the following error:
1) If I use `'diambra-engine>2.1.0rc5'` tests CICD does not work
1) If I use `'diambra-engine>2.1.x'` deploy CICD does not work

Locally both are fine. Could you please help out in this regard?

**Comment by discordianfish** [comment]
Closed in favor of #58

Just for context: We don't need to pin the python version here. Since diambra-engine is generated by grpcio-tools 1.48.2 it will have the constraint `protobuf <4.0dev,>=3.12.0` which means it can by installed with python 3.11 just fine.

Now if you try to install tensorflow as well, it will depend on `protobuf <3.20,>=3.9.2`, limiting the overall constraint to `protobuf <3.20,>=3.12.0`. That works with python 3.10 but in 3.11 it won't find any version since all candidates of protobuf support only <=3.10.

**Pull request closed without merging**